src/Helpers/MemberTokenGenerator.php

```diff
@@ -0,0 +1,76 @@
+<?php declare(strict_types=1);
+
+namespace Firesphere\GraphQLJWT\Helpers;
+
+use App\Users\GraphQL\Types\TokenStatusEnum;
+use InvalidArgumentException;
+use SilverStripe\Core\Extensible;
+use SilverStripe\Security\Member;
+
+/**
+ * Generates / Validates a MemberTokenType for graphql responses
+ *
+ * @mixin Extensible
+ */
+trait MemberTokenGenerator
+{
+    /**
+     * Humanise error message based on status code
+     *
+     * @param string $status
+     * @return string
+     * @throws InvalidArgumentException
+     */
+    public function getErrorMessage(string $status): string
+    {
+        switch ($status) {
+            case TokenStatusEnum::STATUS_EXPIRED:
+                return _t('JWT.STATUS_EXPIRED', 'Token is expired, please renew your token with a refreshToken query');
+            case TokenStatusEnum::STATUS_DEAD:
+                return _t('JWT.STATUS_DEAD', 'Token is expired, but is too old to renew. Please log in again.');
+            case TokenStatusEnum::STATUS_INVALID:
+                return _t('JWT.STATUS_INVALID', 'Invalid token provided');
+            case TokenStatusEnum::STATUS_BAD_LOGIN:
+                return _t('JWT.STATUS_BAD_LOGIN', 'Sorry your email and password combination is rejected');
+            case TokenStatusEnum::STATUS_OK:
+                return _t('JWT.STATUS_OK', 'Token is ok');
+            default:
+                throw new InvalidArgumentException("Invalid status");
+        }
+    }
+
+    /**
+     * Generate MemberToken response
+     *
+     * @param string $status Status code
+     * @param Member $member
+     * @param string $token
+     * @return array Response in format required by MemberToken
+     */
+    protected function generateResponse(string $status, Member $member = null, string $token = null): array
+    {
+        // Success response
+        if ($status === TokenStatusEnum::STATUS_OK) {
```

Comment:
Could you change this to
```
            $response = [
                'Valid'   => false,
                'Member'  => null,
                'Token'   => $token,
                'Status'  => $status,
                'Code'    => 401,
                'Message' => $this->getErrorMessage($status),
            ];
```
followed by the if (omitting the else). As per CodeClimate suggestion:
https://codeclimate.com/github/Firesphere/silverstripe-graphql-jwt/src/Helpers/MemberTokenGenerator.php/source?from_sha=2a45ae2a&to_sha=8fd6b877

Reply:
![image](https://user-images.githubusercontent.com/936064/56869933-ddf68680-6a5b-11e9-9df9-4768f57dfcbe.png)

What?

Reply:
I'll just switch it to a single assignment, and move the conditionals to the line items.

Really weird coding conventions I must exclaim...

Reply:
I don't use an if at all now. :D

Reply:
Apparently this is a thing http://elseisnevernecessary.com